### PR TITLE
doc(changelog): update changelog for 0.6.0

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,364 @@
...
+* The QUIC transport is enabled by default. Furthermore, go-ipfs will automatically run a migration to listen on the QUIC transport (on the same address/port as the TCP transport) to make this upgrade process seamless.
+* The new NOISE security transport is now supported but won't be selected by default. This transport will replace SECIO as the default cross-language interoperability security transport. TLS 1.3 will still remain the default go-ipfs to go-ipfs security transport for now.
+
+**MIGRATION:** This release contains a small config migration to enable QUIC by default. This migration will:
+
+* Normalize multiaddrs in the bootstrap list to use the `/p2p/Qm...` syntax for multiaddrs instead of the `/ipfs/Qm...` syntax.
+* Add QUIC addresses for the default bootstrapers, as necessary. If you've removed the default bootstrappers from your bootstrap config, the migration won't add them back.
+* Add a QUIC listener address to mirror any TCP addresses present in your config. For example, if you're listening on `/ip4/0.0.0.0/tcp/1234`, this migration will add a listen address for `/ip4/0.0.0.0/udp/1234/quic`.
+
+#### QUIC by default
```
Comment (Stebalien) [inline CHANGELOG.md:18]: @marten-seemann LGTY? Additions?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,364 @@
...
+* Because QUIC connections don't consume file descriptors, we're able to remove the rate limit on outbound QUIC connections, further speeding up DHT queries.
+
+Unfortunately, this change isn't without drawbacks: the QUIC transport may not be able to max out some links (usually due to [poorly tuned kernel parameters](https://github.com/lucas-clemente/quic-go/issues/2586#issuecomment-639247615)). On the other hand, it may also be _faster_ in some cases
+
+If you hit this performance issue on Linux, you should tune the `net.core.rmem_default` and `net.core.rmem_max` sysctl parameters to increase your UDP receive buffer sizes.
+
+**NOTE:** The QUIC transport included in this release is backwards incompatible with the experimental QUIC transport included in previous releases. Unfortunately, the QUIC protocol underwent some significant breaking changes and supporting multiple versions wasn't an option. In practice, practice as go-ipfs will simply fall back on the TCP transport when dialing nodes with incompatible QUIC versions.
+
+#### NOISE Transport
```
Comment (Stebalien) [inline CHANGELOG.md:34]: @yusefnapora could you motivate the NOISE transport?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,364 @@
...
+
+* Quick "why noise"?
+
+#### Gateway
+
```
Comment (Stebalien) [inline CHANGELOG.md:39]: cc @lidel

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,364 @@
...
+* Base36 support.
+* 404 pages
+
+#### Gossipsub Upgrade
+
```
Comment (Stebalien) [inline CHANGELOG.md:44]: cc @vyzo

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,364 @@
...
+* Summary paragraph and a link to the blog post?
+
+#### Connectivity
+
```
Comment (Stebalien) [inline CHANGELOG.md:48]: @Stebalien

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,364 @@
...
+* "Peering" support.
+
+#### Libp2p Changes
+
```
Comment (Stebalien) [inline CHANGELOG.md:52]: @stebalien (aarsh is offline)

File: CHANGELOG.md (hunk 1 above)
Comment (marten-seemann) [reply CHANGELOG.md:18]: LGTM.

File: CHANGELOG.md (hunk 4 above)
Comment (vyzo) [reply CHANGELOG.md:44]: You want me to write this or will you do that?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,364 @@
 # go-ipfs changelog
 
+## v0.6.0 2020-06-16
+
+### Highlights
+
+This is a relatively small release in terms of code changes, but it contains some large changes with respect to the protocol. The highlights in this release include:
```
Comment (ribasushi) [inline CHANGELOG.md:7]: Minor reword

```suggestion
This is a relatively small release in terms of code changes, but it contains some significant modifications with respect to protocols. The highlights in this release include:
```

Comment (ribasushi) [inline CHANGELOG.md:7]: Minor reword

```suggestion
This is a relatively small release in terms of code changes, but it contains some significant modifications with respect to protocols. The highlights in this release include:
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,364 @@
...
+
+* The QUIC transport is enabled by default. Furthermore, go-ipfs will automatically run a migration to listen on the QUIC transport (on the same address/port as the TCP transport) to make this upgrade process seamless.
+* The new NOISE security transport is now supported but won't be selected by default. This transport will replace SECIO as the default cross-language interoperability security transport. TLS 1.3 will still remain the default go-ipfs to go-ipfs security transport for now.
```
Comment (ribasushi) [inline CHANGELOG.md:10]:
```suggestion
* The new NOISE security transport is now supported but won't be selected by default. This transport will replace SECIO as the default cross-language interoperability security transport. TLS 1.3 will still remain the default security transport between go-ipfs nodes for now.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,364 @@
...
+
+**MIGRATION:** This release contains a small config migration to enable QUIC by default. This migration will:
```
Comment (ribasushi) [inline CHANGELOG.md:12]:
```suggestion
**MIGRATION:** This release contains a small config migration to enable QUIC alongside the configured TCP transport. This migration will:
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,364 @@
...
+* Because QUIC connections don't consume file descriptors, we're able to remove the rate limit on outbound QUIC connections, further speeding up DHT queries.
+
+Unfortunately, this change isn't without drawbacks: the QUIC transport may not be able to max out some links (usually due to [poorly tuned kernel parameters](https://github.com/lucas-clemente/quic-go/issues/2586#issuecomment-639247615)). On the other hand, it may also be _faster_ in some cases
+
+If you hit this performance issue on Linux, you should tune the `net.core.rmem_default` and `net.core.rmem_max` sysctl parameters to increase your UDP receive buffer sizes.
+
+**NOTE:** The QUIC transport included in this release is backwards incompatible with the experimental QUIC transport included in previous releases. Unfortunately, the QUIC protocol underwent some significant breaking changes and supporting multiple versions wasn't an option. In practice, practice as go-ipfs will simply fall back on the TCP transport when dialing nodes with incompatible QUIC versions.
```
Comment (ribasushi) [inline CHANGELOG.md:32]:
```suggestion
**NOTE:** The QUIC transport included in this release is backwards incompatible with the experimental QUIC transport included in previous releases. Unfortunately, the QUIC protocol underwent some significant breaking changes and supporting multiple versions wasn't an option. In practice this degrades gracefully, as go-ipfs will simply fall back on the TCP transport when dialing nodes with incompatible QUIC versions.
```

**Comment by RubenKelevra** [comment]
@ribasushi I like to do some changes too, but you already recommended changes to those passages :)

File: CHANGELOG.md (hunk 3 above)
Comment (lidel) [reply CHANGELOG.md:39]: Wrote a few words in https://github.com/ipfs/go-ipfs/commit/fa65b71bdc3f492d6f7c3ec803ce0fcf12dd4548 – feel free to edit / let me know if that's enough.

File: CHANGELOG.md (hunk 4 above)
Comment (Stebalien) [reply CHANGELOG.md:56]: Yes, if you have anything to say on the topic (just a headline, really).

File: CHANGELOG.md (hunk 7 above)
Comment (Stebalien) [reply CHANGELOG.md:7]: Does "significant changes to the IPFS protocol" make more sense?

File: CHANGELOG.md (hunk 4 above)
Comment (Stebalien) [reply CHANGELOG.md:69]: Eh, I wrote a small blurb. Feel free to submit a PR to extend it.

**Pull request merged**